sorted_set.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+type sortedSet[T comparable] struct {
```

Comment:
I would just move this to `compression.go`, at the bottom of the file

Reply:
Happy to do this, but extracted it just in case it was thought to be used elsewhere. I originally had it at the bottom.